Arrowgene.Ddon.GameServer/Handler/SkillSetPresetAbilityListHandler.cs

```diff
@@ -0,0 +1,75 @@
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Logging;
+using System.Linq;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class SkillSetPresetAbilityListHandler : GameRequestPacketHandler<C2SSkillSetPresetAbilityListReq, S2CSkillSetPresetAbilityListRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(SkillGetPresetAbilityListHandler));
+
+        private readonly JobManager _jobManager;
+
+
+        public SkillSetPresetAbilityListHandler(DdonGameServer server) : base(server)
+        {
+            _jobManager = server.JobManager;
+        }
+
+        public override S2CSkillSetPresetAbilityListRes Handle(GameClient client, C2SSkillSetPresetAbilityListReq packet)
+        {
+            CDataPresetAbilityParam preset = client.Character.AbilityPresets.Where(x => x.PresetNo == packet.PresetNo).FirstOrDefault();
+
+            if (preset is null)
+            {
+                throw new ResponseErrorException(ErrorCode.ERROR_CODE_SKILL_INVALID_PRESET_NO);
+            }
+
+            CharacterCommon targetCharacter = client.Character;
+            if (packet.PawnId > 0)
+            {
+                targetCharacter = client.Character.Pawns.Where(x => x.PawnId == packet.PawnId).FirstOrDefault();
+
+                if (targetCharacter is null)
+                {
+                    throw new ResponseErrorException(ErrorCode.ERROR_CODE_PAWN_INVALID_SLOT_NO);
+                }
+            }
+
+            _jobManager.SetAbilityPreset(Server.Database, client, targetCharacter, preset);
```

Comment:
Since you are using the new handler, you don't need to save `_jobManager` anymore. Instead you can directly use `Server.JobManager`.